### PR TITLE
feat: polish explore pagination

### LINKED ISSUE #11: feat: add appropriate mobile-first pagination
### Affected area

src\app\[locale]\(explore)\HomePage.tsx, 

### Problem to solve

Currently there is no pagination to the home page in displaying facility cards. This is fine while there are not many facilities, but in the case of more than 50 facilities, there may be issues fetching from the api and displaying it smoothly.

### Proposed solution

There should be appropriate pagination that is also suited for a mobile-first experience. If the contributor is up for it, they can also implement lazy loading for mobile devices. Otherwise, they can implement [shadcn's pagination component](https://ui.shadcn.com/docs/components/pagination) in the home page for an appropriate number of facility cards per page, and also adjust the query to the db to limit the number of facilities queried accordingly.

### Alternatives considered

_No response_

### Out of scope

_No response_

### Acceptance criteria

- [x] Clear UX/UI behavior defined
- [x] Edge cases considered
- [ ] Tests added/updated
- [ ] Documentation updated

### Priority

P0 - Critical

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Muneer320**
# Summary

- Polished the overall pagination logic
- Reworked the explore pagination controls with icon-only previous/next buttons and consistent layouts on both desktop and mobile.
- Tuned the default page size to 10 facilities per page so pagination feels snappier on smaller screens.

## Related Issues / Tickets

- Closes #11

## Contributor info

- LinkedIn profile: https://www.linkedin.com/in/muneer320
<sub>Not a woman, but will be there if I can be of help!<sub>

## Type of change

- [ ] Fix (non-breaking change that fixes an issue)
- [x] Feature (non-breaking change that adds functionality)
- [ ] Refactor (no functional change)
- [ ] Chore/Docs/Build
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How to Test

```bash
npm install
npm run dev
```

1. Launch the app and open the explore page.
2. Verify ten facilities render per page and page transitions use the new icon buttons.
3. Test the mobile viewport (or responsive dev tools) to confirm previous/summary/next alignment and functioning controls.

## Screenshots / Recordings

- Phone Screen
image

- Larger Screens
image

_Ignore any repeated facilities; I included them intentionally for testing purposes._

## Checklist

- [x] I ran the app locally and verified the change
- [ ] I added/updated tests as needed
- [ ] I updated documentation (README, comments, or storybook) as needed
- [ ] I updated environment docs if env vars changed
- [x] No sensitive info (keys, tokens) is committed
- [ ] Database migration required (see “Database Migration” section)

## Breaking Changes

- None

## Database Migration

- None

## Deployment Notes

- None

## Environment Changes

- None

## Additional Context

- Added `lucide-react` for chevron icons used in pagination buttons.
- `npm run check-types` was executed to confirm the TypeScript build remains healthy.

### CODE REVIEW

**Review by natashaannn — CHANGES_REQUESTED** [review]
Hey Muneer, sorry haven't had time to go through line by line, but was wondering how come there are new gaps between the facility cards introduced in the UI? Also can you rebase on the latest main for the CI checks to run? There's also changes to the facility cards in #17 to note also, if that's merged in first will ask you to rebase onto that. I'll do another review once you have made those changes! @missabawse can add on too!

**Comment by Muneer320** [comment]
Hi @natashaannn, 
Sorry for the delayed response, I've been caught up with some other commitments.

I've rebased the branch onto the latest main (commit 384358a), which includes all recent changes:

- ✅ Facility descriptions (PR #17)
- ✅ Operating hours support (PR #34)
- ✅ All other merged fixes
The pagination implementation now works seamlessly with the new operating hours feature. All type checks are passing, and I've tested locally, everything works as expected.

Ready for re-review when you have a moment!

File: todo
Comment (eileenchua) [inline todo]: remove this unused file

File: src/app/[locale]/(explore)/page.tsx [hunk 1]
```diff
@@ -1,141 +1,228 @@
...
+});
+
+type PageProps = {
+  searchParams?: PageSearchParams | Promise<PageSearchParams>;
+};
+
+export default async function Page(props: PageProps) {
+  const searchParams = (await props.searchParams) ?? {};
+  const rawPage = typeof searchParams.page === 'string' ? Number.parseInt(searchParams.page, 10) : 1;
+  const rawPageSize = typeof searchParams.pageSize === 'string' ? Number.parseInt(searchParams.pageSize, 10) : DEFAULT_PAGE_SIZE;
```
Comment (eileenchua) [inline src/app/[locale]/(explore)/page.tsx]: Were you planning to implement page size options?

File: src/app/[locale]/(explore)/page.tsx [hunk 2]
```diff
@@ -1,141 +1,228 @@
...
+      locationsData = locRows.map(normalizeLocation);
+    }
+
+    const facilityTypeIds = Array.from(new Set(facilitiesData.map(f => f.facility_type_id))).filter(Boolean);
+    if (facilityTypeIds.length) {
+      const typeRows = await db.select().from(facilityTypes).where(inArray(facilityTypes.id, facilityTypeIds));
+      facilityTypesData = typeRows.map(normalizeFacilityType);
+    }
+  } else {
+    // Supabase provided data but may have omitted related tables; fill missing pieces locally if available.
```
Comment (eileenchua) [inline src/app/[locale]/(explore)/page.tsx]: do we expect this scenario?

File: src/app/[locale]/(explore)/HomePage.tsx [hunk 3]
```diff
@@ -1,42 +1,84 @@
...
+    });
   }, []);
 
-  // Keep sortedLocations in sync when server data updates after navigation/refresh
   useEffect(() => {
     setSortedLocations(locationsData);
   }, [locationsData]);
 
-  const handleSearch = (address: Address) => {
+  useEffect(() => {
```
Comment (eileenchua) [inline src/app/[locale]/(explore)/HomePage.tsx]: is this required?

**Review by eileenchua — CHANGES_REQUESTED** [review]
Hello @Muneer320, thanks for making the changes, and also for adding the translations! 
I have left some comments below though we will need more time to review the filtering logic. 

Please also check for any `npm run build` errors as the build was failing previously. 

@natashaannn will need your eyes for the filtering logic

**Comment by Muneer320** [comment]
Hello again @eileenchua,
Apologies for the late reply, I've been tied up over the past few days. Thanks for the review! Here are the follow-ups on each comment:

1. **Build verification**
   - `npm run build` now succeeds locally when run with `FORCE_LOCAL_DB=1`, which mimics the CI setup for builds/tests without requiring a live `DATABASE_URL`. Without that flag the build still expects the database URL, so CI just needs one of those two inputs configured.

2. **Unused `todo` file**
   - Removed the stray `todo` file in 77e1dcb to keep the tree clean.

3. **Page-size handling**
   - Keeping the `pageSize` query parsing so QA (and Playwright) can adjust the page length via URL while we iterate on the pagination UX. If you’d prefer to drop that hook entirely I can remove it in a follow-up.

4. **Preserving existing comments**
   - Restored the upstream note above the Supabase fallback block so the intent stays visible. Let me know if there are other sections you’d like called out.

5. **Supabase vs. local fallback**
   - We’ve hit cases in testing where Supabase returns facility rows but omits related tables (e.g. when permissions are tightened), so the conditional local backfill stays to guard against that. Happy to revisit once the API contracts are finalized.

6. **Client-side filtering/search in `HomePage`**
   - The `useEffect` keeps the client copy of locations in sync with the server payload, and the client-side filter gives us instant updates for distance sorting and category changes without another round-trip. If we move all filtering server-side later we can trim this down, but for now it keeps the pagination summary consistent with the visible cards.

Let me know if you need any extra context or prefer alternate tweaks, happy to adjust.

**Comment by eileenchua** [comment]
@Muneer320 No worries, thanks for the update and I appreciate your clear comments on the changes! Previously when I checked the behaviour seems fine, but do give us some time for code review and to look at the DB part as well.

File: src/app/[locale]/(explore)/HomePage.tsx [hunk 4]
```diff
@@ -85,41 +280,103 @@ export default function HomePage({
...
-              userLongitude={userLocation?.longitude ?? 0}
-            />
-          );
-        });
-      })}
+
+      <div className="mt-6">
+        <CategoryScroller selectedCategory={activeCategory} onCategorySelect={handleCategorySelect} />
+        {categoryLabel && (
+          <p className="mt-3 text-base text-muted-foreground">
```
Comment (victoria-lo) [inline src/app/[locale]/(explore)/HomePage.tsx]: Hi @Muneer320 thanks for the PR! Works good on my local and I appreciate the clear separation of UI Pagination components and page logic. Memoization for expensive calculations in HomePage also looks good.

I see this part slightly redundant since the filters are already clearly shown above. Can you remove the part I highlighted in red? 

Image

Probably will look better in mobile if we don't have extra gaps like in the live app now:

Image

File: src/app/[locale]/(explore)/page.tsx (hunk 2 above)
Comment (victoria-lo) [reply src/app/[locale]/(explore)/page.tsx:179]: @Muneer320 thanks for explaining the rationale behind this, makes sense. Can you help add an inline comment above the fallback explaining why it exists and a TODO with the condition for removal (e.g., "remove once API guarantees include related tables")

**Review by victoria-lo — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(explore)/HomePage.tsx (hunk 4 above)
Comment (Muneer320) [reply src/app/[locale]/(explore)/HomePage.tsx]: Do you want the number of facilities to show up, or shall I remove that too?

Comment (Muneer320) [reply src/app/[locale]/(explore)/HomePage.tsx]: For now, I kept the facility indicator, let me know if you want it removed.

**Comment by natashaannn** [comment]
Hey @Muneer320, apologies for the failed CI - we're trying to resolve it in #47 but that's awaiting review. @victoria-lo will also get to addressing your changes shortly - in the mean time you can keep watch of #47 and rebase onto it once it is merged! I'll give a heads up as well!
